js/components/BranchDropdown.react.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+
+// Import the Autocomplete Component
+import Autocomplete from 'react-autocomplete';
+
+export default class BranchDropdown extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+
+    // Set initial State
+    this.state = {
+      // Current value of the select field
+      currentBranch: this.getFirstBranch(),
+      autocompleteData: this.getCompleteData()
+    };
+
+    // Bind `this` context to functions of the class
+    this.onChange = this.onChange.bind(this);
+    this.onSelect = this.onSelect.bind(this);
+    this.getItemValue = this.getItemValue.bind(this);
+    this.renderItem = this.renderItem.bind(this);
+  }
+
+  /**
+   * Callback triggered when the user types in the autocomplete field
+   *
+   * @param {Event} e JavaScript Event
+   * @return {Event} Event of JavaScript can be used as usual.
+   */
+  onChange(e) {
+    console.log('e.target.value ' + e.target.value);
+    console.log('e.target.currentbranch ' + e.target.currentBranch);
+    this.setState({
+      currentBranch: e.target.value
+    });
+  }
+
+  /**
+   * Callback triggered when the autocomplete input changes.
+   *
+   * @param {Object} val Value returned by the getItemValue function.
+   * @return {Nothing} No value is returned
+   */
+  onSelect(val){
+    this.setState({
+      currentBranch: val,
+    });
+
+    this.props.callbackFromParent(val);
+  }
+
+
+  getFirstBranch() {
+    // var branches = this.getBranches();
+    // if (branches.length > 0) {
+    //   return branches[0];
+    // }
+    // return null;
+    return "Show all";
+  }
+
+  /**
```

Comment:
Is this commented out code required?

Reply:
@neomorphic  I just updated the PR and the commented-out code is gone.